locator/abstract_replication_strategy.hh

```diff
@@ -160,10 +164,15 @@ public:
     };
 
 private:
+    using address_ranges = std::unordered_multimap<inet_address, dht::token_range>;
+    using pending_ranges = std::unordered_multimap<range<token>, inet_address>;
+    using pending_ranges_interval_map = boost::icl::interval_map<token, std::unordered_set<inet_address>>;
+
     abstract_replication_strategy::ptr_type _rs;
     token_metadata_ptr _tmptr;
     replication_map _replication_map;
     size_t _replication_factor;
+    pending_ranges_interval_map _pending_ranges_interval_map;
```

Comment:
Idea: add a pointer to the effective replication map for the same nodes, without any pending ranges. Perhaps it can reduce double calculation of things that don't change.

This is not so important here, but will be important if we fold dead nodes into the effective replication map, since these can change often.

Reply:
What do you mean by "same nodes"?
Can you please give an example?

Reply:
Let's say this is the relication map of {n1, n2, n3, n4, n5, pn6}. Add a pointer to the replication map of {n1, n2, n3, n4, n5}. The read qorums of the two are the same.

Similarly, consider the replication map of a cluster with a dead node: {n1, n2, dn3, n4, n5}. Its replication map will be mostly similar to {n1, n2, n3, n4, n5}, and when the node comes back alive it will revert back to it.